remix/app/routes/newsletters.tsx

```diff
@@ -0,0 +1,61 @@
+import type { User } from "@prisma/client";
+import type { LoaderArgs } from "@remix-run/node";
+import {
+  Form,
+  isRouteErrorResponse,
+  useLoaderData,
+  useRouteError,
+} from "@remix-run/react";
+import { SocialsProvider } from "remix-auth-socials";
+import { authenticator } from "~/services/auth.server";
+import type { UserProfile } from "~/types";
+import { db } from "~/utils/db.server";
+
+export const loader = async ({ request }: LoaderArgs) => {
+  const userRequest = (await authenticator.isAuthenticated(request)) as any;
+  const userProfile: UserProfile = userRequest?._json;
+  if (!userProfile) {
+    throw new Response("Forbidden", { status: 403 });
+  }
+  let user: User | null = null;
+
+  try {
+    user = await db.user.findUnique({
```

Comment:
For DB operations, it is best to use the repository pattern by defining a new class for each model/entity:

```
// in app/repositories
import { db } from `~/config/db.server";

class UserRespository {

  static function findBy(param): User {
     //....
  }

  // other functions
}
```

It would allow to `) not having to import `db` in any routes (so abstracting where data is coming) and implement adequate typing/